packages/smooth_app/lib/temp/product_preferences.dart

```diff
@@ -226,60 +265,19 @@ class UserPreferencesModel extends ChangeNotifier {
       return null;
     }
     for (final AttributeGroup attributeGroup in product.attributeGroups) {
+      if (attributeGroup.attributes == null) {
+        continue;
+      }
       for (final Attribute attribute in attributeGroup.attributes) {
         if (attribute.id == attributeId) {
           return attribute;
         }
-        // TODO(monsieurtanuki): have it fixed upstream
+        // TODO(stephanegigandet): have it fixed upstream?
```

Comment:
in fact I don't understand why we have this code. The only place where I see "palm-oil-free" is this commented out line:
user_preferences_model.dart://  static const String ATTRIBUTE_PALM_OIL_FREE = 'palm-oil-free';

And we do want to have const strings for attributes names, then we just have to change it to ATTRIBUTE_PALM_OIL_FREE = 'palm_oil_free';

palm_oil_free is the name of the attribute in the API: https://fr.openfoodfacts.org/api/v0/attribute_groups

Reply:
The problem is that it's `'palm-oil-free'` in each product and `'palm_oil_free'` in the attribute groups json.

Reply:
Ah, my bad. I will fix it server-side. https://github.com/openfoodfacts/openfoodfacts-server/issues/5007

Reply:
The server side fix has been deployed, we can remove this code.